presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -125,54 +127,109 @@ private static Expression toPredicate(Domain domain, QualifiedNameReference refe
+        }
+
+        List<Expression> rangeConjuncts = new ArrayList<>();
+        if (!range.getLow().isLowerUnbounded()) {
+            switch (range.getLow().getBound()) {
+                case ABOVE:
+                    rangeConjuncts.add(new ComparisonExpression(GREATER_THAN, reference, toExpression(range.getLow().getValue(), type)));
+                    break;
+                case EXACTLY:
+                    rangeConjuncts.add(new ComparisonExpression(GREATER_THAN_OR_EQUAL, reference, toExpression(range.getLow().getValue(),
+                            type)));
+                    break;
+                case BELOW:
+                    throw new IllegalStateException("Low Marker should never use BELOW bound: " + range);
+                default:
+                    throw new AssertionError("Unhandled bound: " + range.getLow().getBound());
+            }
+        }
+        if (!range.getHigh().isUpperUnbounded()) {
+            switch (range.getHigh().getBound()) {
+                case ABOVE:
+                    throw new IllegalStateException("High Marker should never use ABOVE bound: " + range);
+                case EXACTLY:
+                    rangeConjuncts.add(new ComparisonExpression(LESS_THAN_OR_EQUAL, reference, toExpression(range.getHigh().getValue(), type)));
+                    break;
+                case BELOW:
+                    rangeConjuncts.add(new ComparisonExpression(LESS_THAN, reference, toExpression(range.getHigh().getValue(), type)));
+                    break;
+                default:
+                    throw new AssertionError("Unhandled bound: " + range.getHigh().getBound());
+            }
+        }
+        // If rangeConjuncts is null, then the range was ALL, which should already have been checked for
+        checkState(!rangeConjuncts.isEmpty());
+        return combineConjuncts(rangeConjuncts);
+    }
+
+    private static Expression combineRangeWithExcludedPoints(Type type, QualifiedNameReference reference, Range range, List<Expression> excludedPoints)
+    {
+        checkArgument(!(excludedPoints.isEmpty() && range.isAll()), "cannot combine empty excluded points list with an ALL range");
+
+        if (!range.isAll() && excludedPoints.isEmpty()) {
+            return processRange(type, range, reference);
+        }
+
+        if (!excludedPoints.isEmpty()) {
+            Expression excludedPointsExpression = new InPredicate(reference, new InListExpression(excludedPoints));
+            if (excludedPoints.size() == 1) {
+                excludedPointsExpression = new ComparisonExpression(EQUAL, reference, getOnlyElement(excludedPoints));
+            }
+
+            if (range.isAll()) {
+                return new NotExpression(excludedPointsExpression);
+            }
+            else {
+                return and(processRange(type, range, reference), new NotExpression(excludedPointsExpression));
```

Comment:
if we use combineConjuncts here, we can blindly processRange and have that return TRUE in the ALL case.